### PR TITLE
added quickfix element for task 233

### PR DESCRIPTION
**Pull request by mareknovotny**
taiga task #231 Basic quick fix support in rules and tooling

### CODE REVIEW

**Comment by mareknovotny** [comment]
Implementation without updating/creating tests

File: graph/api/src/main/java/org/jboss/windup/graph/model/QuickfixModel.java [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    void setName(String name);
+
+    @Property(PROPERTY_NAME)
+    String getName();
+
+    @Property(PROPERTY_SEARCH_STRING)
+    void setSearch(String searchStr);
+
+    @Property(PROPERTY_SEARCH_STRING)
+    String getSearchStr();
```
Comment (jsight) [inline graph/api/src/main/java/org/jboss/windup/graph/model/QuickfixModel.java]: Why is there a "Str" suffix on the getter methods?

File: tooling/api/src/main/java/org/jboss/windup/tooling/data/Classification.java [hunk 2]
```diff
@@ -39,6 +39,12 @@
     Iterable<Link> getLinks();
 
     /**
+     * This contains a list of {@link Quickfix}s for follow up in tools
+     * @return
+     */
+    Iterable<Quickfix> getQuickfixes();
```
Comment (jsight) [inline tooling/api/src/main/java/org/jboss/windup/tooling/data/Classification.java]: Do you think this will be applicable to Classifications as well?

**Comment by jsight** [comment]
This looks good to me so far. I don't think the test needs to be particularly complicated. Just something that shows the handlers parse them correctly and that the quick fix gets put into the graph correctly should be sufficient.

I think you are really close at this point.

File: graph/api/src/main/java/org/jboss/windup/graph/model/QuickfixModel.java (hunk 1 above)
Comment (mareknovotny) [inline graph/api/src/main/java/org/jboss/windup/graph/model/QuickfixModel.java]: copy&paste error, I used that in parsing blocks somewhere else. Good catch!

File: tooling/api/src/main/java/org/jboss/windup/tooling/data/Classification.java (hunk 2 above)
Comment (mareknovotny) [inline tooling/api/src/main/java/org/jboss/windup/tooling/data/Classification.java]: Do you mean the multiple quickfixes or just quickfixes? I thought it is a rare case, but why not.

Actually I wasn't too convinced, but classification was in planning document.

**Comment by mareknovotny** [comment]
Great, I will finish it today (your tomorrow).

File: tooling/api/src/main/java/org/jboss/windup/tooling/data/Classification.java (hunk 2 above)
Comment (jsight) [inline tooling/api/src/main/java/org/jboss/windup/tooling/data/Classification.java]: Oh, I had forgotten that this was in the planning document. I'm not entirely sure what the use case would be, but I guess there is no harm in leaving it here. Maybe that would be suitable for file level replacements (XSLT, etc).

**Comment by jsight** [comment]
Great... merged!

**Pull request merged**